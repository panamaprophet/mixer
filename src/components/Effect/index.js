import React from 'react';

import Fader from '/components/Fader';


const Effect = ({
    name = 'Untitled',
    parameters = [],
    onParamChange = () => {},
}) => {
    return (
        <div className="effect">
            <div className="title">{name}</div>

            {parameters && parameters.map(parameter => (
                <div className="param" key={parameter.id}>
                    <span className="param__title">{parameter.name}:</span>
                    <Fader onChange={onParamChange(parameter.id)} position={parameter.value} />
                </div>
            ))}
        </div>
    );
}

export default Effect;