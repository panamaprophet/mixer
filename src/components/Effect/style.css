.effect {
  margin: 8px auto 0;
  background: #c4cace; /* url("i/metal-texture-small.jpg"); */
  background-size: contain;
  box-shadow: 0 0 3px #000;
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
}

.title {
  font-weight: bold;
  margin-bottom: 16px;
  padding: 8px 16px;
  background: #1b5e20;
  color: rgba(255,255,255,0.87);
  border-radius: 4px 4px 0 0;
}

.effect:not(:first-child) .title {
  background: #827717;
}