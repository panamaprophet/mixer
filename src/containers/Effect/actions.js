import {curry} from 'ramda';


export const onParamChange = curry((dispatch, effectId, parameterId, value) => dispatch({
    type: 'SET_EFFECT_PARAM_VALUE',
    payload: {
        effectId,
        parameterId,
        value,
    }
}));