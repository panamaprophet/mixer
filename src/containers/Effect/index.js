import React, {useContext} from 'react';

import Effect from '/components/Effect';
import Context from '/containers/Context';

import {
    onParamChange,
} from './actions';

const EffectContainer = props => {
    const dispatch = useContext(Context);

    return (
        <Effect 
            {...props} 
            onParamChange={onParamChange(dispatch, props.id)} 
        />
    );
};

export default EffectContainer;