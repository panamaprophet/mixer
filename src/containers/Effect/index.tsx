import React, {useContext} from 'react';

import Effect from '/components/Effect';
import Context from '/containers/Context';

import {
    setSendParamValue,
} from '/store/actions';


const EffectContainer = props => {
    const dispatch = useContext(Context);

    return (
        <Effect
            {...props}
            onParamChange={setSendParamValue(dispatch, props.id)}
        />
    );
};

export default EffectContainer;