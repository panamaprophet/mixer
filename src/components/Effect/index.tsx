import React from 'react';

import EffectParameter from '/components/EffectParameter';

import style from './style.css';


type ParameterType = 'fader' | 'radio';

type Parameter = {
    id: string,
    name: string,
    type: ParameterType,
}

type Props = {
    name: string,
    parameters: Array<Parameter>,
    onParamChange: (id) => void,
};


const Effect = ({
    name = 'Untitled',
    parameters = [],
    onParamChange = () => {},
}: Props) => {
    return (
        <div className={style.effect}>
            <div className={style.title}>{name}</div>

            {parameters && parameters.map(parameter => (
                <div className={style.parameterValue} key={parameter.id}>
                    <EffectParameter {...parameter} onChange={onParamChange} />
                </div>
            ))}
        </div>
    );
}

export default Effect;