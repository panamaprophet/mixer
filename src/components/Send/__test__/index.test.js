import React from 'react';
import {shallow} from 'enzyme';

import Effect from '../';


const propsMock = {
    name: 'Effect',
    parameters: [{
        id: 'effect-parameter-1',
        name: 'Effect Parameter 1',
        value: 42.0,
    }, {
        id: 'effect-parameter-2',
        name: 'Effect Parameter 2',
        value: 71.0,
    }],
    onParamChange: jest.fn(),
};

describe('<Effect />', () => {
    it('should renders without any errors', () => {
        const wrapper = shallow(
            <Effect {...propsMock} />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe('parameters', () => {
        it('should render all passed parameters', () => {
            const wrapper = shallow(
                <Effect {...propsMock} />
            );

            const result = wrapper.find('EffectParameter').length;

            expect(result).toBe(propsMock.parameters.length);
        });
    });

    describe('onParamChange', () => {
        it('should be called when parameter changes', () => {
            const wrapper = shallow(
                <Effect {...propsMock} />
            );

            wrapper.find('EffectParameter').at(0).prop('onChange')();

            expect(propsMock.onParamChange).toBeCalled();
        });
    });
});