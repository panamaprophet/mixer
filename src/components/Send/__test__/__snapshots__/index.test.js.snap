// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Effect /> should renders without any errors 1`] = `
<div
  className="effect"
>
  <div
    className="title"
  >
    Effect
  </div>
  <div
    className="parameterValue"
    key="effect-parameter-1"
  >
    <EffectParameter
      id="effect-parameter-1"
      name="Effect Parameter 1"
      onChange={[MockFunction]}
      value={42}
    />
  </div>
  <div
    className="parameterValue"
    key="effect-parameter-2"
  >
    <EffectParameter
      id="effect-parameter-2"
      name="Effect Parameter 2"
      onChange={[MockFunction]}
      value={71}
    />
  </div>
</div>
`;
